test/plugin/base_test.rb

```diff
@@ -305,20 +305,83 @@ def test_structured_payload_malformatted_log
   def test_structured_payload_json_log
     setup_gce_metadata_stubs
     setup_logging_stubs do
-      d = create_driver
-      json_string = '{"msg": "test log entry 0", "tag2": "test", "data": 5000}'
+      d = create_driver(APPLICATION_DEFAULT_CONFIG)
+      json_string = '{"msg": "test log entry 0", "tag2": "test", ' \
+                    '"data": 5000, "some_null_field": null}'
+      d.emit('message' => 'notJSON ' + json_string)
+      d.emit('log' => 'notJSON ' + json_string)
+      d.emit('msg' => 'notJSON ' + json_string)
+      d.emit('message' => json_string)
+      d.emit('log' => json_string)
+      d.emit('msg' => json_string)
+      d.emit('message' => "  \r\n \t" + json_string)
+      d.emit('log' => "  \r\n \t" + json_string)
+      d.emit('msg' => "  \r\n \t" + json_string)
+      d.run
+    end
+    log_index = 0
+    verify_log_entries(9, COMPUTE_PARAMS, '') do |entry|
+      log_index += 1
+      if log_index % 3 == 1
```

Comment:
Maybe better keep test configuration and test logic separate? I.e. instead of this assertion have a bool flag in the test configuration

Reply:
Good idea. It was getting pretty unwieldy. Done.